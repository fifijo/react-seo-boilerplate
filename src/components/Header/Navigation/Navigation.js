import React, { Fragment } from 'react'
import { Link } from 'react-router-dom'
import './Navigation.scss'

const Navigation = () => {
  return(
    <div className='Navigation'>
      <Link to="/"> Home </Link>
      <Link to="/about"> About </Link>
      <Link to="/contact"> Contact </Link>
    </div>
  )
}
  
export default Navigation