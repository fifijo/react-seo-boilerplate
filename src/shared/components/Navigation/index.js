import React, { Fragment } from 'react'
import { NavLink } from 'react-router-dom'
import styled from 'styled-components'

const StyledNavLink = styled(NavLink)`
  padding: 0 1rem;
  text-decoration: none;
  color: #19233b;
  font-size: 1.2rem;
  font-weight: 700;
  text-transform: uppercase;
  &.active {
    text-decoration: underline;
  }
`;

const Navigation = () => (
    <Fragment>
        <StyledNavLink exact to='/'>Home</StyledNavLink>
        <StyledNavLink to='/about'>About</StyledNavLink>
        <StyledNavLink to='/contact'>Contact</StyledNavLink>
    </Fragment>
)

Navigation.displayName = 'Navigation'

export default Navigation
