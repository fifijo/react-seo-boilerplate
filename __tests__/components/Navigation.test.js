import React from 'react'
import { mount, render } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'
import Navigation from 'shared/components/Navigation'

describe('Navigation component', () => {

    it('should mount correctly', () => {
        expect(mount(
          <Router>
            <Navigation />
          </Router>
        )).toMatchSnapshot() 
    })

    it('should render correctly', () => {
        expect(render(
          <Router>
            <Navigation />
          </Router>
        )).toMatchSnapshot() 
    })
    
  })