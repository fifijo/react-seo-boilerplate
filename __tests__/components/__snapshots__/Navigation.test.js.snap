// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should mount correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Navigation>
      <Styled(NavLink)
        exact={true}
        to="/"
      >
        <StyledComponent
          exact={true}
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "hUpHLV",
                "rules": Array [
                  "
  padding: 0 1rem;
  text-decoration: none;
  color: #19233b;
  font-size: 1.2rem;
  font-weight: 700;
  text-transform: uppercase;
  &.active {
    text-decoration: underline;
  }
",
                ],
              },
              "displayName": "Styled(NavLink)",
              "foldedComponentIds": Array [],
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "NavLink",
                "propTypes": Object {
                  "activeClassName": [Function],
                  "activeStyle": [Function],
                  "aria-current": [Function],
                  "className": [Function],
                  "exact": [Function],
                  "innerRef": [Function],
                  "isActive": [Function],
                  "location": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "strict": [Function],
                  "style": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          to="/"
        >
          <NavLink
            className="sc-bdVaJa hUpHLV"
            exact={true}
            to="/"
          >
            <Link
              aria-current="page"
              className="sc-bdVaJa hUpHLV active"
              style={Object {}}
              to={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": null,
                }
              }
            >
              <LinkAnchor
                aria-current="page"
                className="sc-bdVaJa hUpHLV active"
                href="/"
                navigate={[Function]}
                style={Object {}}
              >
                <a
                  aria-current="page"
                  className="sc-bdVaJa hUpHLV active"
                  href="/"
                  onClick={[Function]}
                  style={Object {}}
                >
                  Home
                </a>
              </LinkAnchor>
            </Link>
          </NavLink>
        </StyledComponent>
      </Styled(NavLink)>
      <Styled(NavLink)
        to="/about"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "hUpHLV",
                "rules": Array [
                  "
  padding: 0 1rem;
  text-decoration: none;
  color: #19233b;
  font-size: 1.2rem;
  font-weight: 700;
  text-transform: uppercase;
  &.active {
    text-decoration: underline;
  }
",
                ],
              },
              "displayName": "Styled(NavLink)",
              "foldedComponentIds": Array [],
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "NavLink",
                "propTypes": Object {
                  "activeClassName": [Function],
                  "activeStyle": [Function],
                  "aria-current": [Function],
                  "className": [Function],
                  "exact": [Function],
                  "innerRef": [Function],
                  "isActive": [Function],
                  "location": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "strict": [Function],
                  "style": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          to="/about"
        >
          <NavLink
            className="sc-bdVaJa hUpHLV"
            to="/about"
          >
            <Link
              aria-current={null}
              className="sc-bdVaJa hUpHLV"
              to={
                Object {
                  "hash": "",
                  "pathname": "/about",
                  "search": "",
                  "state": null,
                }
              }
            >
              <LinkAnchor
                aria-current={null}
                className="sc-bdVaJa hUpHLV"
                href="/about"
                navigate={[Function]}
              >
                <a
                  aria-current={null}
                  className="sc-bdVaJa hUpHLV"
                  href="/about"
                  onClick={[Function]}
                >
                  About
                </a>
              </LinkAnchor>
            </Link>
          </NavLink>
        </StyledComponent>
      </Styled(NavLink)>
      <Styled(NavLink)
        to="/contact"
      >
        <StyledComponent
          forwardedComponent={
            Object {
              "$$typeof": Symbol(react.forward_ref),
              "attrs": Array [],
              "componentStyle": ComponentStyle {
                "componentId": "sc-bdVaJa",
                "isStatic": false,
                "lastClassName": "hUpHLV",
                "rules": Array [
                  "
  padding: 0 1rem;
  text-decoration: none;
  color: #19233b;
  font-size: 1.2rem;
  font-weight: 700;
  text-transform: uppercase;
  &.active {
    text-decoration: underline;
  }
",
                ],
              },
              "displayName": "Styled(NavLink)",
              "foldedComponentIds": Array [],
              "propTypes": Object {
                "activeClassName": [Function],
                "activeStyle": [Function],
                "aria-current": [Function],
                "className": [Function],
                "exact": [Function],
                "innerRef": [Function],
                "isActive": [Function],
                "location": [Function],
                "onClick": [Function],
                "replace": [Function],
                "strict": [Function],
                "style": [Function],
                "target": [Function],
                "to": [Function],
              },
              "render": [Function],
              "styledComponentId": "sc-bdVaJa",
              "target": Object {
                "$$typeof": Symbol(react.forward_ref),
                "displayName": "NavLink",
                "propTypes": Object {
                  "activeClassName": [Function],
                  "activeStyle": [Function],
                  "aria-current": [Function],
                  "className": [Function],
                  "exact": [Function],
                  "innerRef": [Function],
                  "isActive": [Function],
                  "location": [Function],
                  "onClick": [Function],
                  "replace": [Function],
                  "strict": [Function],
                  "style": [Function],
                  "target": [Function],
                  "to": [Function],
                },
                "render": [Function],
              },
              "toString": [Function],
              "warnTooManyClasses": [Function],
              "withComponent": [Function],
            }
          }
          forwardedRef={null}
          to="/contact"
        >
          <NavLink
            className="sc-bdVaJa hUpHLV"
            to="/contact"
          >
            <Link
              aria-current={null}
              className="sc-bdVaJa hUpHLV"
              to={
                Object {
                  "hash": "",
                  "pathname": "/contact",
                  "search": "",
                  "state": null,
                }
              }
            >
              <LinkAnchor
                aria-current={null}
                className="sc-bdVaJa hUpHLV"
                href="/contact"
                navigate={[Function]}
              >
                <a
                  aria-current={null}
                  className="sc-bdVaJa hUpHLV"
                  href="/contact"
                  onClick={[Function]}
                >
                  Contact
                </a>
              </LinkAnchor>
            </Link>
          </NavLink>
        </StyledComponent>
      </Styled(NavLink)>
    </Navigation>
  </Router>
</BrowserRouter>
`;

exports[`Navigation component should render correctly 1`] = `
Array [
  <a
    aria-current="page"
    class="sc-bdVaJa hUpHLV active"
    href="/"
  >
    Home
  </a>,
  <a
    class="sc-bdVaJa hUpHLV"
    href="/about"
  >
    About
  </a>,
  <a
    class="sc-bdVaJa hUpHLV"
    href="/contact"
  >
    Contact
  </a>,
]
`;
